### PR TITLE
Refactored TRLocationController to swift.

### PR DESCRIPTION
**Pull request by katsuroo**
Co-authored-by: Adam Sharp <adam@sharplet.me>

### CODE REVIEW

File: Sources/TroposCore/Controllers/LocationController.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    @objc public override convenience init() {
+        if #available(iOSApplicationExtension 10.0, *) {
+            dispatchPrecondition(condition: .onQueue(.main))
+        }
+
+        self.init(locationManager: CLLocationManager())
+    }
+
+    @objc public func authorizationStatusEqualTo(_ status: CLAuthorizationStatus) -> Bool {
```
[the remark this answers is not in the document]
Comment (katsuroo) [reply Sources/TroposCore/Controllers/LocationController.swift]: This was declared as a public interface in the objective-c file so I think it is still being used elsewhere.

[the remark this answers is not in the document]
Comment (sharplet) [reply Sources/TroposCore/Controllers/LocationController.swift]: Ah I can see now that it’s used in `TRApplicationController`.

File: Sources/TroposCore/Controllers/LocationController.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                case .restricted, .denied:
+                    return false
+                case .notDetermined:
+                    return nil
+                }
+            }.take(first: 1)
+
+            isAuthorized.start(observer)
+
+            if authorizationStatus.value == .notDetermined {
```
[the remark this answers is not in the document]
Comment (katsuroo) [reply Sources/TroposCore/Controllers/LocationController.swift]: Makes sense. Just read up a little on swift asynchronous code execution.

File: Sources/TroposIntents/CheckWeatherIntentHandler.swift [hunk 3]
```diff
@@ -17,8 +17,8 @@ public final class CheckWeatherIntentHandler: NSObject, CheckWeatherIntentHandli
         let geocode = GeocodeController()
         let location = LocationController()
 
-        let weatherUpdate = location.requestAlwaysAuthorization()
-            .flatMap(.latest) { _ in location.updateCurrentLocation() }
+        let weatherUpdate = location.requestAuthorization()
+            .flatMap(.latest) { _ in location.requestLocation() }
```
[the remark this answers is not in the document]
Comment (sharplet) [reply Sources/TroposIntents/CheckWeatherIntentHandler.swift]: True, I got a little confused about the fact that both names still exist. Agreed we can remove the extension methods later.

File: Sources/TroposCore/Controllers/LocationController.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                case .authorizedAlways, .authorizedWhenInUse:
+                    return true
+                case .restricted, .denied:
+                    return false
+                case .notDetermined:
+                    return nil
+                }
+            }.take(first: 1)
+
+            isAuthorized.start(observer)
```
[the remark this answers is not in the document]
Comment (katsuroo) [reply Sources/TroposCore/Controllers/LocationController.swift]: Added in latest commit

**Pull request merged**